cli/commands/info/index.ts

```diff
@@ -0,0 +1,116 @@
+    assertExists(agentRegistry, 'agentRegistry')
+    assertExists(agentRegistryContractAddress, 'agentRegistryContractAddress')
+    assertExists(getFromIpfs, 'getFromIpfs')
+    assertExists(getTransactionReceipt, 'getTransactionReceipt')
+
+    return async function info(daysToScan = DEFAULT_DAYS_TO_SCAN) {
+        const finalAgentId = args.agentId ? args.agentId : agentId;
+
+        assertIsNonEmptyString(finalAgentId, 'agentId');
+
+        const agent = await agentRegistry.getAgent(finalAgentId);
+        const currentState = await agentRegistry.isEnabled(finalAgentId) as boolean
+        const ipfsMetaHash = agent.metadata;
+
+
+        const ipfsData = await getFromIpfs(ipfsMetaHash)
+        printIpfsMetaData(ipfsData,currentState)
+
+        console.log(`Recent Activity (Last ${daysToScan} days): \n`);
+
+        const blockEventTopicFilters = AGENT_REGESTRY_EVENT_FRAGMENTS
+            .filter(fragment => isRelevantSmartContractEvent(fragment.name))
+            .map(eventFragment => getTopicHashFromEventName(eventFragment.name as StateChangeContractEvent)) as string[];
+
+        const latestBlockNumber = await ethersAgentRegistryProvider.getBlockNumber();
+
+        const network = await ethersAgentRegistryProvider.getNetwork();
+        const { chainId } = network;
+        const { blockTimeInSeconds } = getBlockChainNetworkConfig(chainId);
+
+        console.log(`Days to scan is: ${daysToScan}, Block to subtract is: ${(daysToScan * SECONDS_IN_DAY)/(blockTimeInSeconds)}`)
+        const endingBlock = Math.floor(latestBlockNumber - ((daysToScan * SECONDS_IN_DAY)/(blockTimeInSeconds)));
+
+        const increment = 1000;
+        let startingBlock = latestBlockNumber - (increment * 5);
+
+        const getAgentLogs = async (startBlock: number, endBlock: number) => {
+            return await ethersAgentRegistryProvider.getLogs({
+                address: agentRegistryContractAddress,
+                fromBlock: startBlock,
+                toBlock: endBlock,
+                topics: [[...blockEventTopicFilters], null] // This assumes that each smart contract event is indexed so that the first topic is always an event of interest
+            });
+        }
+        
+        console.log(`Starting block: ${startingBlock} and ending Block: ${endingBlock}`)
+        while(startingBlock > endingBlock) {
+            
+            // Get logs in parallel from 5 diffrent block ranges
+            const response = (await Promise.all([
+                getAgentLogs(startingBlock, startingBlock + increment), 
+                getAgentLogs(startingBlock + increment + 1, startingBlock + (increment * 2)),
+                getAgentLogs(startingBlock + (2 * increment) + 1, startingBlock + (increment * 3)),
+                getAgentLogs(startingBlock + (3 * increment) + 1, startingBlock + (increment * 4)),
+                getAgentLogs(startingBlock + (4 * increment) + 1, startingBlock + (increment * 5))
+            ]))
```

Comment:
should be a way to automate this code block

Reply:
Could you expand on what you mean by this?

Reply:
i was thinking of using the `Array.map` function to reduce some of the repeated code (which can be error-prone) i.e.
```
const responses = await Promise.all([0, 1, 2, 3, 4, 5].map(i => getAgentLogs(startingBlock + (i*increment), startingBlock + ((i+1)*increment))))
```